### PR TITLE
Custom Hitsounds

### PR DESCRIPTION
**Pull request by Oly-UK**
How to use:
1. Download hitsounds, make sure they are '.wav' files (https://anonfile.com/Z73fUapbod/hitsound_zip)
2. Locate csgo/sounds, example -> C:\Program Files (x86)\Steam\steamapps\common\Counter-Strike Global Offensive\csgo\sound
3. Drag your hitsounds into that folder
4. Select "Custom" from the hitsound combobox
5. type the name of the hitsound you wish to use, **do not** include the extension

Video:
https://streamable.com/nxgr4h

### CODE REVIEW

**Comment by notgoodusename** [comment]
Its cool and all but what about the volume?

**Comment by Oly-UK** [comment]
> Its cool and all but what about the volume?

yeah i tried doing this https://www.unknowncheats.me/forum/counterstrike-global-offensive/256149-adjusting-playsound-volume.html but i couldn't get it to work

**Comment by danielkrupinski** [comment]
No need to use `PlaySoundA` function - **csgo/sound** directory is mapped by game so sound files put there can be played using `play` command - just like standard hitsound does.

File: Osiris/Config.h [hunk 1]
```diff
@@ -269,6 +269,7 @@ class Config {
         bool fixTabletSignal{ false };
         float maxAngleDelta{ 255.0f };
         bool fakePrime{ false };
+        char customhitsound[50]{ "" };
```
Comment (danielkrupinski) [inline Osiris/Config.h:272]: Use std::string instead

**Comment by Oly-UK** [comment]
> Use `play` command instead of `PlaySoundA` function.

you're suggestion to use the play command doesn't work

**Comment by Oly-UK** [comment]
oh yes i'm stupid okay i have done as requested, uploading now :)

**Comment by Oly-UK** [comment]
@danielkrupinski done :D

**Comment by Oly-UK** [comment]
also mind checking out my other pull requests? @danielkrupinski #1436 #1437

File: Osiris/Hacks/Misc.cpp [hunk 2]
```diff
@@ -12,6 +12,24 @@
...
+void Misc::BetterHitsounds(GameEvent* event) noexcept
+{
+    if (!config->misc.hitSound)
+        return;
+    std::string soundFile = "play ";
+    soundFile += config->misc.customhitsound;
+    soundFile += ".wav";
+    if (config->misc.hitSound == 5)
+    {
+        if (interfaces->engine->getPlayerForUserID(event->getInt("attacker")) == interfaces->engine->getLocalPlayer() || interfaces->engine->getPlayerForUserID(event->getInt("userid")) == interfaces->engine->getLocalPlayer())
```
Comment (playday3008) [inline Osiris/Hacks/Misc.cpp:28]: `Engine::getLocalPlayer` is `deprecate`

Comment (playday3008) [reply Osiris/Hacks/Misc.cpp:28]: Use `localPlayer->index()` instead of `interfaces->engine->getLocalPlayer()`

**Comment by Oly-UK** [comment]
@playday3008 done ;)

**Comment by danielkrupinski** [comment]
Why don't modify `Misc::playHitSound` function to handle custom sound instead of creating another function? Please remove "winmm.lib" dependency.

**Comment by Oly-UK** [comment]
@danielkrupinski yeah idk why i did that tbh, will do this in a moment

**Comment by Oly-UK** [comment]
@danielkrupinski pushed both requests, haven't tested but should be working

**Comment by danielkrupinski** [comment]
Thanks for your pull request! I've decided to implement your idea in slightly different way.

**Pull request closed without merging**